my-react-study-app/src/4_widgets/mainSection/mainSection.tsx

```diff
@@ -0,0 +1,81 @@
+import { Outlet, NavLink, useSearchParams } from 'react-router-dom';
+
+import './mainSection.css';
+import { CharecterRequest, EpisodeRequest, LocationRequest, MainSectionProps } from '../../7_shared/types';
+import Pagination from '../pagination/pagination';
+
+export default function MainSection(props: MainSectionProps) {
+    const { header, items, info, handleRequest } = props;
+    const [, setSearchParams] = useSearchParams();
+
+    const changePage = (page: number) => {
+        setSearchParams({ page: page.toString() });
+    };
+
+    return (
+        <main className="main">
+            <div className="main__wrapper">
+                <div className="main__content">
+                    <div className="main__search_section">
+                        <div className="main__items">
+                            {Object.keys(info).length !== 0 && (
+                                <Pagination info={info} handleRequest={handleRequest} onClick={changePage} />
+                            )}
+                            {header === '' &&
+                                items.map((item, index) => {
+                                    const elem = item as [string, string];
+                                    return (
+                                        <div className="main__item" key={`${String(index)}`}>
+                                            <p className="main__item_text">{elem[0].slice(0, -1)}</p>
+                                        </div>
+                                    );
+                                })}
+                            {header.includes('location') &&
+                                items.map((item, index) => {
+                                    const location = item as LocationRequest;
+                                    return (
+                                        <NavLink
+                                            to={`location/detail/${location.id}`}
+                                            className="main__item"
+                                            key={`${String(index)}`}
```

Comment:
It's a bad aproach to use index as key. It' better to use id  or other unique values

Reply:
also `${index}` already change type to string, you can skip String